translator/translate/otel/receiver/otlp/translator.go

```diff
@@ -56,18 +59,20 @@ func WithDataType(dataType component.DataType) Option {
 var _ common.Translator[component.Config] = (*translator)(nil)
 
 func NewTranslator(opts ...Option) common.Translator[component.Config] {
-	t := &translator{factory: otlpreceiver.NewFactory()}
-	for _, opt := range opts {
-		opt.apply(t)
-	}
-	return t
+	return NewTranslatorWithName("", opts...)
 }
 
 func NewTranslatorWithName(name string, opts ...Option) common.Translator[component.Config] {
-	t := &translator{name: name, factory: otlpreceiver.NewFactory()}
+	t := &translator{name: name, instanceNum: -1, factory: otlpreceiver.NewFactory()}
 	for _, opt := range opts {
 		opt.apply(t)
 	}
+	if name == "" && t.dataType != "" {
+		t.name = string(t.dataType)
+		if t.instanceNum != -1 {
```

Comment:
I see instanceNum being set to -1 above but when is it not that?
Do we have a test to ensure having multiple otlp sections in the same config works as you are expecting it to here?

Reply:
Added the unit test